src/Microsoft.ApplicationInsights.AspNet/TelemetryInitializers/ComponentVersionTelemetryInitializer.cs

```diff
@@ -0,0 +1,38 @@
+﻿namespace Microsoft.ApplicationInsights.AspNet.TelemetryInitializers
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Channel;
+    using DataContracts;
+    using Microsoft.AspNet.Http;
+    using Microsoft.Extensions.Configuration;
+
+    /// <summary>
+    /// A telemetry initializer that populates telemetry.Context.Component.Version to the value read from project.json
+    /// </summary>
+    public class ComponentVersionTelemetryInitializer : TelemetryInitializerBase
+    {
+        private const string versionConfigurationOption = "dependencies:Microsoft.ApplicationInsights.AspNet";
+
+        public ComponentVersionTelemetryInitializer(IHttpContextAccessor httpContextAccessor):base(httpContextAccessor)
```

Comment:
Can you get IConfiguration object using dependency injection here and than try to do: 
```
config[versionConfigurationOption].ToString();
```
instead of guessing which file contains configuration?